addons/pos_coupon/static/src/js/coupon.js

```diff
@@ -0,0 +1,1053 @@
+     * @emits 'rewards-updated' emitted after 'update-rewards' callback.
+     */
+    var _order_super = models.Order.prototype;
+    models.Order = models.Order.extend({
+        // OVERIDDEN METHODS
+
+        initialize: function () {
+            let res = _order_super.initialize.apply(this, arguments);
+            res.on(
+                'update-rewards',
+                () => {
+                    if (!this.pos.config.use_coupon_programs) return;
+                    dp.add(this._getNewRewardLines()).then(([newRewardLines, rewardsContainer]) => {
+                        this.orderlines.remove(this._getRewardLines());
+                        this.orderlines.add(newRewardLines);
+                        // We need this for the rendering of ActivePrograms component.
+                        this.rewardsContainer = rewardsContainer;
+                        // Send a signal that the rewardsContainer are updated.
+                        this.trigger('rewards-updated');
+                    });
+                },
+                res
+            );
+            res.on('reset-coupons', res.resetCoupons, res);
+            res._initializePrograms();
+            return res;
+        },
+        init_from_JSON: function (json) {
+            _order_super.init_from_JSON.apply(this, arguments);
+            this.bookedCouponCodes = json.bookedCouponCodes;
+            this.activePromoProgramIds = json.activePromoProgramIds;
+        },
+        export_as_JSON: function () {
+            let json = _order_super.export_as_JSON.apply(this, arguments);
+            return Object.assign(json, {
+                bookedCouponCodes: this.bookedCouponCodes,
+                activePromoProgramIds: this.activePromoProgramIds,
+            });
+        },
+        set_orderline_options: function (orderline, options) {
+            _order_super.set_orderline_options.apply(this, [orderline, options]);
+            if (options && options.is_program_reward) {
+                orderline.is_program_reward = true;
+                orderline.tax_ids = options.tax_ids;
+                orderline.program_id = options.program_id;
+                orderline.coupon_id = options.coupon_id;
+            }
+        },
+        /**
+         * This function's behavior is modified so that the reward lines are
+         * rendered at the bottom of the orderlines list.
+         */
+        get_orderlines: function () {
+            return [...this._getRegularOrderlines(), ...this._getRewardLines()];
+        },
+        _getRegularOrderlines: function () {
+            const orderlines = _order_super.get_orderlines.apply(this, arguments);
+            return orderlines.filter((line) => !line.is_program_reward);
+        },
+        _getRewardLines: function () {
+            const orderlines = _order_super.get_orderlines.apply(this, arguments);
+            return orderlines.filter((line) => line.is_program_reward);
+        },
+        wait_for_push_order: function () {
```

Comment:
it probably should be true only if there are coupon and/or program on orders. Also you should call super, and return result || your condition like in l10n_fr__pos_cert